src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_integration_test.py

```diff
@@ -662,3 +664,119 @@ def test_merge_reductions(self, mock_get_timeout):
+        weight=1.0).put()
+
+    # Set up a Job
+    data_types.Job(
+        environment_string=(
+            'CUSTOM_BINARY = True\n'
+            'FUCHSIA_RESOURCES_URL = gs://fuchsia-on-clusterfuzz-v2/*\n'
+            'QUEUE_OVERRIDE=FUCHSIA\n'
+            'OS_OVERRIDE=FUCHSIA'),
+        name='libfuzzer_asan_test_fuzzer',
+        platform='FUCHSIA',
+        templates=[u'libfuzzer', u'engine_asan']).put()
+
+    # Set up a JobTemplate
+    data_types.JobTemplate(
+        name='libfuzzer',
+        environment_string=('APP_NAME = launcher.py\n'
+                            'MAX_FUZZ_THREADS = 1\n'
+                            'MAX_TESTCASES = 4\n'
+                            'FUZZ_TEST_TIMEOUT = 4800\n'
+                            'TEST_TIMEOUT = 30\n'
+                            'WARMUP_TIMEOUT = 30\n'
+                            'BAD_BUILD_CHECK = False\n'
+                            'THREAD_ALIVE_CHECK_INTERVAL = 1\n'
+                            'REPORT_OOMS_AND_HANGS = True\n'
+                            'CORPUS_FUZZER_NAME_OVERRIDE = libFuzzer\n'
+                            'ENABLE_GESTURES = False\n'
+                            'THREAD_DELAY = 30.0')).put()
+
+    # Set up another JobTemplate
+    data_types.JobTemplate(
+        name='engine_asan',
+        environment_string=(
+            'LSAN = True\n'
+            'ADDITIONAL_ASAN_OPTIONS = quarantine_size_mb=64:strict_memcmp=1'
+            ':symbolize=0:fast_unwind_on_fatal=0'
+            ':allocator_release_to_os_interval_ms=500\n')).put()
+
+    environment.set_value('QUEUE_OVERRIDE', 'FUCHSIA')
+    environment.set_value('OS_OVERRIDE', 'FUCHSIA')
+    os.environ['FUCHSIA_RESOURCES_URL'] = 'gs://fuchsia-on-clusterfuzz-v2/*'
+    # set_bot_environment gives us access to RESOURCES_DIR
+    environment.set_bot_environment()
+    resources_dir = environment.get_value('RESOURCES_DIR')
+    if not resources_dir:
+      raise Exception('Could not find RESOURCES_DIR')
+    fuchsia_resources_dir = os.path.join(resources_dir, 'fuchsia')
+    pkey_path = os.path.join(fuchsia_resources_dir, '.ssh', 'pkey')
+    portnum = '56339'
+    environment.set_value('FUCHSIA_PKEY_PATH', pkey_path)
+    environment.set_value('FUCHSIA_PORTNUM', portnum)
+
+  def tearDown(self):
+    environment.set_value('QUEUE_OVERRIDE', '')
+    environment.set_value('OS_OVERRIDE', '')
+
+  def test_fuzzer_can_boot_and_run(self):
+    """Tests running a single round of fuzzing on a Fuchsia target, using
+    'echo' in place of a fuzzing command."""
+
+    # Cannot call setUp from our class's setUp method, because we're using the
+    # cloud emulation decorator, which assumes the base class is
+    # unittest.TestCase, and recurses infinitely if this is not true (as in our
+    # case, with BaseLauncherTest as the base)
+    # TODO(flowerhack): Does it make more sense to factor out all the setUp
+    # into its own method, which will have to be specifically called by every
+    # class in this file?
```

Comment:
This is a question for y'all; I'm neutral either way, since neither route is super-aesthetically-pleasing for me

Reply:
I'm confused, why is there a need to call setUp manually?
If you want to share it across multiple classes maybe use a mixin?

Reply:
Right, maybe my phrasing wasn't super clear:

* the with_cloud_emulators wrapper wraps a class, in this case TestLauncherFuchsia, and calls its setUp method

* but, it does not call any further parent's setUp methods! (e.g. BaseLauncherTest, which is what we want)

* and, if you try to call super(self).setUp within TestLauncherFuchsia, it winds up recursing infinitely, since you're just calling the decorator's setUp method

Do you prefer a mixin to simply being explicit when calling setUp while using this decorator?

(I moved this line into setUp but it's still the same line fyi)